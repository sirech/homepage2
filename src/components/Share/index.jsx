import React from 'react'
import {
  FacebookShareButton,
  FacebookIcon,
  TwitterShareButton,
  TwitterIcon,
  LinkedinShareButton,
  LinkedinIcon,
  EmailShareButton,
  EmailIcon,
} from 'react-share'

import { path as Rpath } from 'ramda'

import frontmatterType from '../../prop-types/frontmatter'
import siteType from '../../prop-types/site'

import styles from './style.module.scss'

const Share = ({ frontmatter, site }) => {
  const { path, title, categories, description } = frontmatter

  const siteUrl = Rpath(['siteMetadata', 'url'])(site)
  const twitter = Rpath(['siteMetadata', 'twitter'])(site)

  const url = siteUrl + path

  return (
    <div className={styles.share}>
      <div className={styles.cta}>Did you like it? Give it a share!</div>
      <div className={styles.buttons}>
        <FacebookShareButton url={url} quote={description}>
          <FacebookIcon size={32} round />
        </FacebookShareButton>

        <TwitterShareButton
          url={url}
          title={title}
          via={twitter.replace(/@/, '')}
          hashtags={categories.map((tag) =>
            tag.toLowerCase().replace(/ /, '-')
          )}
        >
          <TwitterIcon size={32} round />
        </TwitterShareButton>

        <LinkedinShareButton url={url} title={title} summary={description}>
          <LinkedinIcon size={32} round />
        </LinkedinShareButton>

        <EmailShareButton subject={title} body={description + ' ' + url}>
          <EmailIcon size={32} round />
        </EmailShareButton>
      </div>
    </div>
  )
}

Share.propTypes = {
  frontmatter: frontmatterType,
  site: siteType,
}

export default Share
