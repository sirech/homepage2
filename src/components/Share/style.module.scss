.share {
  text-align: center;
}

.cta {
  font-style: italic;
  margin-bottom: 0.3rem;
}

.buttons {
  button:not(:first-child) {
    margin-left: 0.3rem;
  }
}
