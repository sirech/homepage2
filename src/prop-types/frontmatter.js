import PropTypes from 'prop-types'

export default PropTypes.shape({
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  date: PropTypes.string.isRequired,
  layout: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  draft: PropTypes.bool.isRequired,
  description: PropTypes.string,
}).isRequired
