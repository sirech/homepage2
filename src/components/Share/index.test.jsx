import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Share from './index'

import frontmatter from '../../fixtures/frontmatter'
import siteMetadata from '../../fixtures/siteMetadata'

describe('components', () => {
  describe('Share', () => {
    let site

    beforeEach(() => {
      site = {
        siteMetadata: siteMetadata(),
      }
    })

    it('renders the share buttons', () => {
      const component = shallow(
        <Share site={site} frontmatter={frontmatter()} />
      )
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
