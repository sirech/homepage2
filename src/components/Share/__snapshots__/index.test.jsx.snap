// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Share renders the share buttons 1`] = `
<div
  className="share"
>
  <div
    className="cta"
  >
    Did you like it? Give it a share!
  </div>
  <div
    className="buttons"
  >
    <ForwardRef(ShareButton-facebook)
      quote="this is the description"
      url="http://example.com/the-post"
    >
      <Icon
        bgStyle={Object {}}
        borderRadius={0}
        iconFillColor="white"
        round={true}
        size={32}
      />
    </ForwardRef(ShareButton-facebook)>
    <ForwardRef(ShareButton-twitter)
      hashtags={
        Array [
          "javascript",
          "react",
        ]
      }
      title="My Post"
      url="http://example.com/the-post"
      via="handle"
    >
      <Icon
        bgStyle={Object {}}
        borderRadius={0}
        iconFillColor="white"
        round={true}
        size={32}
      />
    </ForwardRef(ShareButton-twitter)>
    <ForwardRef(ShareButton-linkedin)
      summary="this is the description"
      title="My Post"
      url="http://example.com/the-post"
    >
      <Icon
        bgStyle={Object {}}
        borderRadius={0}
        iconFillColor="white"
        round={true}
        size={32}
      />
    </ForwardRef(ShareButton-linkedin)>
    <ForwardRef(ShareButton-email)
      body="this is the description http://example.com/the-post"
      subject="My Post"
    >
      <Icon
        bgStyle={Object {}}
        borderRadius={0}
        iconFillColor="white"
        round={true}
        size={32}
      />
    </ForwardRef(ShareButton-email)>
  </div>
</div>
`;
